### PR TITLE
TF-933 [PART-3] Apply new design for search

### PR DESCRIPTION
**Pull request by dab246**
#933 

### Resolved
- [x] Simplify advanced search request
- [x] Replace `hasKeyword` with `text` in advanced search
- [x] Fill color text search result for query advanced search 

https://user-images.githubusercontent.com/80730648/190575753-46cd4fdb-aa04-4c92-ad9e-b97735f8c4e4.mov

### CODE REVIEW

**Comment by hoangdat** [comment]
Hi @dab246 please change the Advanced Form, All mailbox is the default option for `Mailbox`.

**Comment by dab246** [comment]
> Hi @dab246 please change the Advanced Form, All mailbox is the default option for `Mailbox`.

I updated it. Please review again

File: model/lib/extensions/email_filter_condition_extension.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        hasAttachment != null ||
+        text?.isNotEmpty == true ||
+        from?.isNotEmpty == true ||
+        to?.isNotEmpty == true ||
+        cc?.isNotEmpty == true ||
+        bcc?.isNotEmpty == true ||
+        subject?.isNotEmpty == true ||
+        body?.isNotEmpty == true ||
+        header?.isNotEmpty == true;
+  }
```
Comment (hoangdat) [inline model/lib/extensions/email_filter_condition_extension.dart:27]: @ManhNTX please check this. I remember we have a problem when combine `inMialbox` with other?

**Review by ManhNTX — APPROVED** [review]
LGTM

File: model/lib/extensions/email_filter_condition_extension.dart (hunk 1 above)
Comment (ManhNTX) [reply model/lib/extensions/email_filter_condition_extension.dart:27]: > @ManhNTX please check this. I remember we have a problem when combine `inMialbox` with other?

@hoangdat BE had fixed this. Don't worry about this.

**Pull request merged**